sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/util/TimestampFormatterSuite.scala

```diff
@@ -433,4 +433,35 @@ class TimestampFormatterSuite extends DatetimeFormatterSuite {
       assert(formatter.format(date(1970, 4, 10)) == "100")
     }
   }
+
+  test("SPARK-31939: Fix Parsing day of year when year field pattern is missing") {
+    // resolved to queryable LocaleDate or fail directly
+    val f0 = TimestampFormatter("yyyy-dd-DD", UTC, isParsing = true)
```

Comment:
shall we test both timestamp and date formatter? We can put the test in the base class.

Reply:
Sounds good.